dev/TreeView/TreeViewList.cpp

```diff
@@ -99,6 +99,14 @@ void TreeViewList::OnContainerContentChanging(const winrt::IInspectable& /*sende
     if (!args.InRecycleQueue())
     {
         auto targetItem = args.ItemContainer().as<winrt::TreeViewItem>();
+
+        // "Re-evaluate" the item source binding to trigger a TreeViewItem.OnPropertChanged(ItemsSourceProperty) changed call.
+        if (auto itemsSource = targetItem.ItemsSource())
```

Comment:
Maybe we could rename this new method to `UpdateItemsSource` to be consistent with the `UpdateIndentation` and `UpdatesSelectionVisual` that already exist.

Reply:
Already done in commit https://github.com/microsoft/microsoft-ui-xaml/pull/2058/commits/d4a1f079982f129f679be35043d902ae0dd914a8 🙂 New function is named "SetItemsSource" for now but I was also thinking about "UpdateItemsSource" for the name. TreeViewItem also has functions to get/set the ItemsSource (TreeViewItem.ItemSource()) so introducing a new function with "Set" in the name might be confusing (there also is no "GetItemsSource"). Will update name to "UpdateItemsSource".

Reply:
I agree with @japf here, I think it may be better to make the function public and call it, instead of relying on the property changed event to call the function.

Reply:
Already done in commit https://github.com/microsoft/microsoft-ui-xaml/commit/d4a1f079982f129f679be35043d902ae0dd914a8 🙂

Reply:
I'm wondering if this API (`SetItemsSource`) needs to be public. From the "outside", when using the tree view control you should not need to deal with the underlying `TreeViewList` control. That looks like an implementation details to me.

Reply:
TreeViewItem needs to expose this function so that it can be called by TreeViewList. You don't have access to this function outside of WinUI (i.e. a project consuming WinUI) as it is not part of the IDL file. Can you clarify what you mean here?

Reply:
You're right, if that's not part of the IDL that's fine :-)
Thanks for clarifying.